--- mpi.io.protos.txt	2014-11-11 19:57:46.264092592 +0800
+++ mpi.io.protos.txt2	2014-11-11 19:57:20.914659683 +0800
@@ -1,7 +1,7 @@
 @@@  MPI I/O Routines
 int MPI_File_close (MPI_File *fh)
 
-int MPI_File_open (MPI_Comm comm, char *filename, int amode, MPI_Info info, MPI_File *fh)
+int MPI_File_open (MPI_Comm comm, const char *filename, int amode, MPI_Info info, MPI_File *fh)
 
 int MPI_File_preallocate (MPI_File fh, MPI_Offset size)
 
@@ -13,11 +13,11 @@
 
 int MPI_File_seek (MPI_File fh, MPI_Offset offset, int whence)
 
-int MPI_File_set_view (MPI_File fh, MPI_Offset disp, MPI_Datatype etype, MPI_Datatype filetype, char *datarep, MPI_Info info)
+int MPI_File_set_view (MPI_File fh, MPI_Offset disp, MPI_Datatype etype, MPI_Datatype filetype, const char *datarep, MPI_Info info)
 
-int MPI_File_write (MPI_File fh, void *buf, int count, MPI_Datatype datatype, MPI_Status *status)
+int MPI_File_write (MPI_File fh, const void *buf, int count, MPI_Datatype datatype, MPI_Status *status)
 
-int MPI_File_write_all (MPI_File fh, void *buf, int count, MPI_Datatype datatype, MPI_Status *status)
+int MPI_File_write_all (MPI_File fh, const void *buf, int count, MPI_Datatype datatype, MPI_Status *status)
 
-int MPI_File_write_at (MPI_File fh, MPI_Offset offset, void *buf, int count, MPI_Datatype datatype, MPI_Status *status)
+int MPI_File_write_at (MPI_File fh, MPI_Offset offset, const void *buf, int count, MPI_Datatype datatype, MPI_Status *status)
 
